### PR TITLE
Throw exceptions in RankingBattle action

### PR DESCRIPTION
**Pull request by boscohyun**
The title is the content.

### CODE REVIEW

**Review by ipdae — COMMENTED** [review]
It would be nice to add a test case.

File: Lib9c/Action/RankingBattle.cs [hunk 1]
```diff
@@ -78,31 +74,27 @@ public override IAccountStateDelta Execute(IActionContext context)
...
-                    context,
-                    "Aborted as the avatar state of the opponent ({OpponentAddress}) was failed to load.",
-                    EnemyAddress
-                );
+                throw new FailedLoadStateException($"Aborted as the avatar state of the opponent ({EnemyAddress}) was failed to load.");
             }
 
             var weeklyArenaState = states.GetWeeklyArenaState(WeeklyArenaAddress);
 
             //FIXME 오류던지게 고쳐야함
```
Comment (limebell) [inline Lib9c/Action/RankingBattle.cs:82]: We can remove this FIXME :)

File: Lib9c/Action/NotEnoughFungibleAssetValueException.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    [Serializable]
+    public class NotEnoughFungibleAssetValueException : Exception
+    {
+        public NotEnoughFungibleAssetValueException(string message) : base(message)
+        {
+        }
+
+        public NotEnoughFungibleAssetValueException(string require, string current)
+            : this(
+                $"Aborted as the signer's balance ({current}) is insufficient to pay entrance fee/stake ({require}).")
```
Comment (limebell) [inline Lib9c/Action/NotEnoughFungibleAssetValueException.cs:18]: How about this for consistency?
```suggestion
                $"Aborted as the signer's balance is insufficient to pay entrance fee/stake: {current} < {require}.")
```

**Review by limebell — APPROVED** [review]

**Review by Unengine — APPROVED** [review]

**Review by ipdae — APPROVED** [review]

File: Lib9c/Action/RankingBattle.cs [hunk 3]
```diff
@@ -78,36 +73,39 @@ public override IAccountStateDelta Execute(IActionContext context)
...
 
             if (!arenaInfo.Active)
             {
-                FungibleAssetValue agentBalance = states.GetBalance(ctx.Signer, states.GetGoldCurrency());
+                FungibleAssetValue agentBalance = default;
+                try
+                {
+                    agentBalance = states.GetBalance(ctx.Signer, states.GetGoldCurrency());
+                }
+                catch (InvalidOperationException)
```
Comment (longfin) [inline Lib9c/Action/RankingBattle.cs:105]: (it isn't critical, but just checking)

Does `states.GetBalance()` throw `InvalidOperationException` ? according to [docs](https://docs.libplanet.io/master/api/Libplanet.Action.IAccountStateDelta.html#Libplanet_Action_IAccountStateDelta_GetBalance_Libplanet_Address_Libplanet_Currency_), it doesn't seem to throw any exceptions.

cc @planetarium/libplanet

Comment (boscohyun) [reply Lib9c/Action/RankingBattle.cs:105]: Found during testing. If I put an address that has never received `Balance` in `states.GetBalance()` as the first parameter, `InvalidOperationException ` occurred.

**Pull request merged**